src/main/java/com/google/firebase/remoteconfig/Version.java

```diff
@@ -0,0 +1,225 @@
+    this.versionNumber = null;
+    this.updateTime = 0L;
+    this.updateOrigin = null;
+    this.updateType = null;
+    this.updateUser = null;
+    this.rollbackSource = null;
+    this.legacy = false;
+  }
+
+  Version(@NonNull VersionResponse versionResponse) {
+    checkNotNull(versionResponse);
+    this.versionNumber = versionResponse.getVersionNumber();
+
+    if (!Strings.isNullOrEmpty(versionResponse.getUpdateTime())) {
+      try {
+        this.updateTime = RemoteConfigUtil.convertToMilliseconds(versionResponse.getUpdateTime());
+      } catch (ParseException e) {
+        throw new IllegalStateException("Unable to parse update time.", e);
+      }
+    } else {
+      this.updateTime = 0L;
+    }
+
+    this.updateOrigin = versionResponse.getUpdateOrigin();
+    this.updateType = versionResponse.getUpdateType();
+    TemplateResponse.UserResponse userResponse = versionResponse.getUpdateUser();
+    this.updateUser = (userResponse != null) ? new User(userResponse) : null;
+    this.description = versionResponse.getDescription();
+    this.rollbackSource = versionResponse.getRollbackSource();
+    this.legacy = versionResponse.isLegacy();
+  }
+
+  /**
+   * Creates a new {@link Version} with a description.
+   */
+  public static Version withDescription(String description) {
+    return new Version().setDescription(description);
+  }
+
+  /**
+   * Gets the version number of the template.
+   *
+   * @return The version number or null.
+   */
+  @Nullable
+  public String getVersionNumber() {
+    return versionNumber;
+  }
+
+  /**
+   * Gets the update time of the version. The timestamp of when this version of the Remote Config
+   * template was written to the Remote Config backend.
+   *
+   * @return The update time of the version or null.
+   */
+  @Nullable
+  public long getUpdateTime() {
+    return updateTime;
+  }
+
+  /**
+   * Gets the origin of the template update action.
+   *
+   * @return The origin of the template update action or null.
+   */
+  @Nullable
+  public String getUpdateOrigin() {
+    return updateOrigin;
+  }
+
+  /**
+   * Gets the type of the template update action.
+   *
+   * @return The type of the template update action or null.
+   */
+  @Nullable
+  public String getUpdateType() {
+    return updateType;
+  }
+
+  /**
+   * Gets the update user of the template.
+   * An aggregation of all metadata fields about the account that performed the update.
+   *
+   * @return The update user of the template or null.
+   */
+  @Nullable
+  public User getUpdateUser() {
+    return updateUser;
+  }
+
+  /**
+   * Gets the user-provided description of the corresponding Remote Config template.
+   *
+   * @return The description of the template or null.
+   */
+  @Nullable
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Gets the rollback source of the template.
+   *
+   * <p>The version number of the Remote Config template that has become the current version
```

Comment:
I'm not 100% clear here.  Are the number and the source the same thing?  Or does this actually get the version from which this version was rolled back?

Reply:
The `rollback source` is a property of the version (metadata) of a rolled back template. It only presents in the Version if the template was a result of a rolled back operation. `rollback source` is actually the version number of the original template that was used for the roll back.